### PR TITLE
update CI to use GitHub Actions

### PR DESCRIPTION
**Pull request by cpanato**
This PRs does the following
- Github Actions per issue: https://github.com/crossplane-contrib/provider-github/issues/9
- changes the go project path to be `github.com/crossplane-contrib/provider-github`  instead of `github.com/crossplane/provider-github`
- Adds issue/pr templates
- Removes JenkinsFile

fixes: https://github.com/crossplane-contrib/provider-github/issues/9

/assign @hasheddan

### CODE REVIEW

**Comment by cpanato** [comment]
Jenkins will not work here 😄  because we removed the file, so will not report back

**Comment by cpanato** [comment]
And need to set up the necessary keys in the secrets as well

**Review by hasheddan — APPROVED** [review]
Awesome! Thanks so much @cpanato :) I will make sure to get the appropriate secrets added. One nitpick below but not super critical.

File: .gitignore [hunk 1]
```diff
@@ -1,9 +1,17 @@
+
 /.cache
 /.work
 /_output
 cover.out
 /vendor
 /.vendor-new
+.registry
```
Comment (hasheddan) [inline .gitignore:8]: This can be dropped as it is from legacy crossplane logic :)

Comment (cpanato) [reply .gitignore]: thanks for the review Dan!
PTAL

**Pull request merged**